src/librbd/Operations.cc

```diff
@@ -1486,24 +1486,21 @@ int Operations<I>::metadata_set(const std::string &key,
     return -EROFS;
   }
 
-  C_SaferCond metadata_ctx;
-  {
-    std::shared_lock owner_lock{m_image_ctx.owner_lock};
-    r = prepare_image_update(exclusive_lock::OPERATION_REQUEST_TYPE_GENERAL,
-                             true);
-    if (r < 0) {
-      return r;
-    }
-
-    execute_metadata_set(key, value, &metadata_ctx);
-  }
+  r = invoke_async_request("metadata_set",
+                           exclusive_lock::OPERATION_REQUEST_TYPE_GENERAL,
+                           true,
```

Comment:
@dillaman  Oops, I think there is mistake here, it should be **false**? So the python API could not permit to update metadata via snapshot.

Reply:
I don't think it was restricted before (was it?) since the metadata applies to both snapshots and the image HEAD revision.

Reply:
@dillaman  Yeah, it was.
We noticed that if the journaling feature is enabled, the event that update metadata is recorded.
However, the event is not recorded if updated via snapshot. Is this going to be a problem?
Thank you.

Reply:
Ah, ok. Can you open a new PR to address this?

Reply:
Thx.
New PR: https://github.com/ceph/ceph/pull/37417